config/v1/types_ingress.go

```diff
@@ -41,13 +44,103 @@ type IngressSpec struct {
 	// certificate.
 	// +optional
 	AppsDomain string `json:"appsDomain,omitempty"`
+
+	// ComponentRoutes is a list of routes configured by a cluster-admin.
```

Comment:
lower-case all these field references. They refer to the JSON field names, not the Golang variable names.

Reply:
"... optional configured by the cluster-admin to override the defaults".